File: app/src/main/java/net/vornao/ddns/dome/devices/Thermostat.java
package net.vornao.ddns.dome.devices;

import androidx.annotation.NonNull;

import net.vornao.ddns.dome.handler.DeviceHandler;

import java.util.Locale;

public class Thermostat extends Device{

    private int mode;
    private float threshold;
    private float temp;
    private float hum;

    public Thermostat(int id, String type, DeviceHandler handler, String home_id, String name) {
        super(id, type, handler, home_id, name);
    }

    public void setThermostat(int mode, float temp){
        String msg = String.format(Locale.ENGLISH, "{\"mode\":%d, \"threshold\":%.1f}", mode, temp);
        this.mode = mode;
        this.threshold = temp;
        handler.publish(pubTopic, msg, 2);
    }

    public void setTemp(float temp){
        this.temp = temp;
    }


    public int getMode() {
        return mode;
    }

    public void setMode(int mode) {
        this.mode = mode;
    }

    public float getThreshold() {
        return threshold;
    }

    public void setThreshold(float threshold) {
        this.threshold = threshold;
    }

    public float getTemp() {
        return temp;
    }

    public float getHum() {
        return hum;
    }

    @NonNull
    @Override
    public String toString() {
        return "Thermostat{" +
                "mode=" + mode +
                ", threshold=" + threshold +
                ", temp=" + temp +
                ", hum=" + hum +
                '}';
    }

    public void setHum(float hum) {
        this.hum = hum;
    }

}

File: app/src/main/java/net/vornao/ddns/dome/devices/Thermostat.kt
package net.vornao.ddns.dome.devices
import net.vornao.ddns.dome.handler.DeviceHandler
import java.util.*

class Thermostat(id: Int, type: String, handler: DeviceHandler, home_id: String, name: String) : Device(id, type, handler, home_id, name) {
    var mode = 0
    var threshold = 0f
    var temp = 0f
    var hum = 0f

    fun setThermostat(mode: Int, temp: Float) {
        val msg = String.format(Locale.ENGLISH, "{\"mode\":%d, \"threshold\":%.1f}", mode, temp)
        this.mode = mode
        threshold = temp
        handler.publish(pubTopic, msg, 2)
    }

    override fun toString(): String {
        return "Thermostat{" +
                "mode=" + mode +
                ", threshold=" + threshold +
                ", temp=" + temp +
                ", hum=" + hum +
                '}'
    }
}